homeassistant/components/plaato/entity.py

```diff
@@ -0,0 +1,103 @@
+"""PlaatoEntity class."""
+from pyplaato.models.device import PlaatoDevice
+
+from homeassistant.helpers import entity
+
+from .const import (
+    DEVICE,
+    DEVICE_ID,
+    DEVICE_NAME,
+    DEVICE_TYPE,
+    DOMAIN,
+    SENSOR_DATA,
+    SENSOR_SIGNAL,
+)
+
+
+class PlaatoEntity(entity.Entity):
+    """Representation of a Plaato Entity."""
+
+    def __init__(self, data, sensor_type, coordinator=None):
+        """Initialize the sensor."""
+        self._coordinator = coordinator
+        self._entry_data = data
+        self._sensor_type = sensor_type
+        self._device_id = data[DEVICE][DEVICE_ID]
+        self._device_type = data[DEVICE][DEVICE_TYPE]
+        self._device_name = data[DEVICE][DEVICE_NAME]
+        self._state = 0
+
+    @property
+    def _attributes(self) -> dict:
+        return PlaatoEntity._to_snake_case(self._sensor_data.attributes)
+
+    @property
+    def _sensor_name(self) -> str:
+        return self._sensor_data.get_sensor_name(self._sensor_type)
+
+    @property
+    def _sensor_data(self) -> PlaatoDevice:
+        if self._coordinator:
+            return self._coordinator.data
+        return self._entry_data[SENSOR_DATA]
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return f"{DOMAIN} {self._device_type} {self._device_name} {self._sensor_name}".title()
+
+    @property
+    def unique_id(self):
+        """Return the unique ID of this sensor."""
+        return f"{self._device_id}_{self._sensor_type}"
+
+    @property
+    def device_info(self):
+        """Get device info."""
+        device_info = {
+            "identifiers": {(DOMAIN, self._device_id)},
+            "name": self._device_name,
+            "manufacturer": "Plaato",
+            "model": self._device_type,
+        }
+
+        if self._sensor_data.firmware_version != "":
+            device_info["sw_version"] = self._sensor_data.firmware_version
+
+        return device_info
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes of the monitored installation."""
+        if self._attributes is not None:
+            return self._attributes
```

Comment:
Please set the state attribute keys explicitly so we can see what they are and the formatting. Eg use a dict that maps between state attribute key and plato data key, and iterate over that fetching the values from the plato data.